tests/rptest/tests/node_pool_migration_test.py

```diff
@@ -0,0 +1,278 @@
+
+
+class NodePoolMigrationTest(PreallocNodesTest):
+    """
+    Basic nodes decommissioning test.
+    """
+    def __init__(self, test_context):
+        self._topic = None
+
+        super(NodePoolMigrationTest, self).__init__(test_context=test_context,
+                                                    num_brokers=10,
+                                                    node_prealloc_count=1)
+
+    def setup(self):
+        # defer starting redpanda to test body
+        pass
+
+    @property
+    def admin(self):
+        # retry on timeout and service unavailable
+        return Admin(self.redpanda, retry_codes=[503, 504])
+
+    def _create_topics(self, replication_factors=[1, 3]):
+        """
+        :return: total number of partitions in all topics
+        """
+        total_partitions = 0
+        topics = []
+        for i in range(10):
+            partitions = random.randint(1, 10)
+            spec = TopicSpec(
+                partition_count=partitions,
+                replication_factor=random.choice(replication_factors))
+            topics.append(spec)
+            total_partitions += partitions
+
+        for spec in topics:
+            self.client().create_topic(spec)
+
+        self._topic = random.choice(topics).name
+
+        return total_partitions
+
+    # after node was removed the state should be consistent on all other not removed nodes
+    def _check_state_consistent(self, decommissioned_id):
+
+        not_decommissioned = [
+            n for n in self.redpanda.started_nodes()
+            if self.redpanda.node_id(n) != decommissioned_id
+        ]
+
+        def _state_consistent():
+
+            for n in not_decommissioned:
+                cfg_status = self.admin.get_cluster_config_status(n)
+                brokers = self.admin.get_brokers(n)
+                config_ids = sorted([s['node_id'] for s in cfg_status])
+                brokers_ids = sorted([b['node_id'] for b in brokers])
+                self.logger.info(
+                    f"brokers: {brokers_ids}, from config: {config_ids}")
+                if brokers_ids != config_ids:
+                    return False
+                if decommissioned_id in brokers_ids:
+                    return False
+
+            return True
+
+        wait_until(_state_consistent, 90, 1)
+
+    def _wait_for_node_removed(self, node_id):
+        waiter = NodeDecommissionWaiter(self.redpanda,
+                                        node_id,
+                                        self.logger,
+                                        progress_timeout=60)
+        waiter.wait_for_removal()
+
+    def _wait_for_nodes_removed(self, decommissioned_ids):
+        threads = []
+        for id in decommissioned_ids:
+            t = Thread(target=lambda: self._wait_for_node_removed(id),
+                       daemon=True)
+            t.start()
+            threads.append(t)
+
+        for t in threads:
+            t.join()
```

Comment:
think join() just ignores any exception from _wait_for_node_removed() and moves on? We should check that the removal is actually successful?